srcs/semaphore.cc

```diff
@@ -80,40 +75,38 @@ void Semaphore::New(const Nan::FunctionCallbackInfo<v8::Value>& info)
     return Nan::ThrowError("Semaphore() expects a boolean as fifth argument");
   if (!info[5]->IsUndefined() && !info[5]->IsUint32())
     return Nan::ThrowError("Semaphore() expects an integer as sixth argument");
-  strict = info[1]->BooleanValue();
-  debug = info[2]->BooleanValue();
-  silent = info[3]->BooleanValue();
-  retry_on_eintr = info[4]->BooleanValue();
-  value = !info[5]->IsUndefined()? info[5]->IntegerValue(): 1;
-  v8::String::Utf8Value str(info[0]->ToString());
-  if (str.length() >= 255 || str.length() <= 0)
-    return Nan::ThrowError("Semaphore() : first argument's length must be < 255 && > 0");
-  buf = (char*)(*str);
-  Semaphore* obj = new Semaphore(buf, str.length(), strict, debug, silent, retry_on_eintr, value);
+  strict = Nan::To<bool>(info[1]).FromJust();
+  debug = Nan::To<bool>(info[1]).FromJust();
+  silent = Nan::To<bool>(info[1]).FromJust();
+  retry_on_eintr = Nan::To<bool>(info[1]).FromJust();
+  value = !info[5]->IsUndefined()? info[5]->IntegerValue(context).FromJust(): 1;
+  v8::String::Utf8Value v8str(isolate, info[0]);
+  std::string str(*v8str);
+  strcpy(buf, str.c_str());
+
+  size_t str_len;
+  str_len = str.length();
+  if (str_len >= SEMSIZE || str_len <= 0)
+  {
+    char msg[100];
+    sprintf(msg, "Semaphore(): first argument's length must be < %i && > 0", SEMSIZE);
+    return Nan::ThrowError(msg);
+  }
+
+  Semaphore* obj = new Semaphore(buf, str_len, strict, debug, silent, retry_on_eintr, value);
   obj->Wrap(info.This());
   info.GetReturnValue().Set(info.This());
 }
 
 void Semaphore::Acquire(const Nan::FunctionCallbackInfo<v8::Value>& info)
 {
-  int        r;
+  int r;
   Semaphore* obj = ObjectWrap::Unwrap<Semaphore>(info.Holder());
 
-  /*if (!obj->strict && (obj->closed || obj->locked))
```

Comment:
Is there a reason to remove this block? Maybe it was left out after your were done debugging?

Reply:
Ah sorry, failed to see that it was commented out.